tests/sanity/extra/botmeta.py

```diff
@@ -54,184 +54,193 @@
-    all_maintainers = filedata['maintainers'] + filedata['ignore']
-    if not filename.startswith(('plugins/action/', 'plugins/doc_fragments/', 'plugins/filter/', 'plugins/module_utils/', 'plugins/plugin_utils/')):
-        maintainers = read_authors(filename)
-        for maintainer in maintainers:
-            maintainer = extract_author_name(maintainer)
-            if maintainer is not None and maintainer not in all_maintainers:
-                others = ', '.join(all_maintainers)
-                msg = f'Author {maintainer} not mentioned as active or inactive maintainer for {filename} (mentioned are: {others})'
-                print(f'{FILENAME}:0:0: {msg}')
-    should_have_no_maintainer = filename in IGNORE_NO_MAINTAINERS
-    if not all_maintainers and not should_have_no_maintainer:
-        print(f'{FILENAME}:0:0: No (active or inactive) maintainer mentioned for {filename}')
-    if all_maintainers and should_have_no_maintainer:
-        print(f'{FILENAME}:0:0: Please remove {filename} from the ignore list of {sys.argv[0]}')
-
-
-def main():
+        author = data.get('author') or []
+        if isinstance(author, str):
+            author = [author]
+        return author
+
+    def extract_author_name(self, author: str) -> str | None:
+        m = self.author_regex.match(author)
+        if m:
+            return m.group(1)
+        if author == 'Ansible Core Team':
+            return '$team_ansible_core'
+        return None
+
+    def validate(self, filename: str, filedata: dict) -> None:
+        if not filename.startswith('plugins/'):
+            return
+        if filename.startswith(('plugins/doc_fragments/', 'plugins/module_utils/')):
+            return
+        # Compile list of all active and inactive maintainers
+        all_maintainers = filedata['maintainers'] + filedata['ignore']
+        if not filename.startswith(('plugins/action/', 'plugins/doc_fragments/', 'plugins/filter/', 'plugins/module_utils/', 'plugins/plugin_utils/')):
+            maintainers = self.read_authors(filename)
+            for maintainer in maintainers:
+                maintainer = self.extract_author_name(maintainer)
+                if maintainer is not None and maintainer not in all_maintainers:
+                    others = ', '.join(all_maintainers)
+                    msg = f'Author {maintainer} not mentioned as active or inactive maintainer for {filename} (mentioned are: {others})'
+                    self.report_error(f'{self.botmeta_filename}:0:0: {msg}')
+        should_have_no_maintainer = filename in IGNORE_NO_MAINTAINERS
+        if not all_maintainers and not should_have_no_maintainer:
+            self.report_error(f'{self.botmeta_filename}:0:0: No (active or inactive) maintainer mentioned for {filename}')
+        if all_maintainers and should_have_no_maintainer:
+            self.report_error(f'{self.botmeta_filename}:0:0: Please remove {filename} from the ignore list of {sys.argv[0]}')
```

Comment:
Hi @felixfontein I will take a closer look tomorrow, today it's just a quick glance. I don't claim to fully understand what's going on here, I have not delved much into that part of the machinery. That being said, I _think_ that referring to `sys.argv[0]` here might no longer make sense.

Reply:
Thanks for taking a look! Yes, you're definitely right, I'll fix that later today.

Reply:
Actually `sys.argv[0]` still works, it will return `tests/sanity/extra/botmeta.py`. I've removed it in f236deda98039a9ef7f33b94c1a9d1daf99c8e1c nontheless.